SynMongoDB.pas

```diff
@@ -4100,6 +4100,23 @@ function TBSONVariant.TryJSONToVariant(var JSON: PUTF8Char;
     end;
     result := true;
   end;
+  procedure Return(kind: shortint; P: PUTF8Char; GotoEndOfObject: AnsiChar='}');overload;

```

Comment:
I got the idea.
This is IMHO a regression of trunk, since the previous code is fine with Delphi, since its early days.
Some small refactoring after merge.